### PR TITLE
Add unit tests for org.wso2.ballerinalang.compiler.util.NumericLiteralSupport

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
I'll add PR text later.

### CODE REVIEW

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    Assert.assertEquals("3", NumericLiteralSupport.stripDiscriminator("3"));
+    Assert.assertEquals("abc", NumericLiteralSupport.stripDiscriminator("abc"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcf"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcF"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcd"));
+    Assert.assertEquals("abc",
+        NumericLiteralSupport.stripDiscriminator("abcD"));
```
Comment (Braavos6) [inline compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: I think for testng you have it the other way. So `Assert.assertEquals(actual,expected)`

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  @Test
+  public void testIsHexLiteral() {
+    Assert.assertTrue(NumericLiteralSupport.isHexLiteral("p\u0000p"));
+    Assert.assertTrue(NumericLiteralSupport.isHexLiteral("P\u0000P"));
+    Assert.assertFalse(NumericLiteralSupport.isHexLiteral("1"));
+  }
+
+  @Test
+  public void testParseBigDecimal() throws Exception {
```
Comment (Braavos6) [inline compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: I don't think you need the Exception as the method being test is not throwing anything. Same for the methods below

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    Assert.assertEquals("1.0",
+        NumericLiteralSupport.parseBigDecimal(1d).toString());
+    Assert.assertEquals("1.0",
+        NumericLiteralSupport.parseBigDecimal(1D).toString());
+    Assert.assertEquals("1",
+        NumericLiteralSupport.parseBigDecimal(1).toString());
```
Comment (Braavos6) [inline compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: Pass a value with a length longer than 2 without a `d` init
Maybe a alphanumeric ie `b1r` and a numeric value

**Review by EricHetti — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java (hunk 3 above)
Comment (JohnLBergqvist) [reply compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: I'm testing for that functionality within the next test for `isDecimalDiscriminated`

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java (hunk 2 above)
Comment (JohnLBergqvist) [reply compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: Done

File: compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java (hunk 1 above)
Comment (JohnLBergqvist) [reply compiler/ballerina-lang/src/test/java/org/wso2/ballerinalang/compiler/util/NumericLiteralSupportTest.java]: You're right. That's annoying.

**Review by EricHetti — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**